frontend/src/app/services/guard/authorization/authorization-guard.service.ts

```diff
@@ -0,0 +1,80 @@
+import {Injectable} from '@angular/core';
+import {ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree} from '@angular/router';
+import {Observable} from 'rxjs';
+import {of} from 'rxjs/internal/observable/of';
+import {AuthenticationService} from '../../authentication/authentication.service';
+import {LoginRedirectService} from '../../authentication/login-redirect.service';
+import {filter} from 'rxjs/internal/operators/filter';
+import {switchMap, tap} from 'rxjs/operators';
+import {User} from '../../../shared/model/User';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AuthorizationGuardService implements CanActivate {
+    constructor(
+        protected router: Router,
+        protected authenticationService: AuthenticationService,
+        protected loginRedirect: LoginRedirectService
+    ) {}
+
+    canActivate(
+        route: ActivatedRouteSnapshot,
+        state: RouterStateSnapshot
+    ): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+        return this.authenticationService.user.pipe(
+            // Only run when fetching of the user object is complete.
+            filter((user: User) => user != null),
+            // Map the user observable on a boolean.
+            switchMap(() => {
+                let authorized = false;
+
+                const authorizations: string[][] = route.data.guards;
+
+                if (authorizations && authorizations.length) {
+                    for (const authorization of authorizations) {
+                        let currentAuthorized = true;
+
+                        for (const guard of authorization) {
+                            currentAuthorized = currentAuthorized && this.hasGuard(
+                                guard
+                            );
+                        }
```

Comment:
```suggestion
const currentAuthorized = authorization.every(guard => this.hasGuard(guard));
```
Is equivalent and much shorter I believe

Reply:
True, fixed!